macOSUpgrade.sh

```diff
@@ -147,10 +147,48 @@ osinstallersetupdDaemonSettingsFilePath="/Library/LaunchDaemons/com.jamfps.clean
 ##Launch agent settings for filevault authenticated reboots
 osinstallersetupdAgentSettingsFilePath="/Library/LaunchAgents/com.apple.install.osinstallersetupd.plist"
 
+##Amount of time (in seconds) to allow a user to connect to AC power before moving on
+##If null or 0, then the user will not have the opportunity to connect to AC power
+acPowerWaitTimer="0"
+
+##Declare the sysRequirementErrors array
+declare -a sysRequirementErrors
+
+##Icon to display during the AC Power warning
+warnIcon="/System/Library/CoreServices/CoreTypes.bundle/Contents/Resources/AlertCautionIcon.icns"
+
+##Icon to display when errors are found
+errorIcon="/System/Library/CoreServices/CoreTypes.bundle/Contents/Resources/AlertStopIcon.icns"
+
 # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
 # FUNCTIONS
 # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
 
+kill_process() {
+    processPID="$1"
+    if /bin/ps -p "$processPID" > /dev/null ; then
+        /bin/kill "$processPID"
+        wait "$processPID" 2>/dev/null
+    fi
+}
+
+wait_for_ac_power() {
+    # Loop for "acPowerWaitTimer" seconds until either AC Power is detected or the timer is up
+    /bin/echo "Waiting for AC power..."
+    while [[ "$acPowerWaitTimer" -gt "0" ]]; do
+        if /usr/bin/pmset -g ps | /usr/bin/grep "AC Power" > /dev/null ; then
+            /bin/echo "Power Check: OK - AC Power Detected"
+            kill_process "$jamfHelperPowerPID"
+            return
+        fi
+        sleep 1
+        ((acPowerWaitTimer--))
+    done
+    kill_process "$jamfHelperPowerPID"
+    sysRequirementErrors+=("• Is connected to AC power")
```

Comment:
It looks like that this is data about the error.
I would say it is better to include the character of `• ` in the contents.

Reply:
Just to clarify, are you referring to printing the bullet like this:
```bash
sysRequirementErrors+=("Is connected to AC power")
$( /usr/bin/printf '\t%s\n' "• ${sysRequirementErrors[@]}" )
```
Rather than including the bullet in the array element itself?

Reply:
Never mind, the above comment won't work. How do you propose?

Reply:
Added to [8de80ff](https://github.com/kc9wwh/macOSUpgrade/pull/115/commits/8de80ff7c2da443a958c2975741e6c3112f27250).